node/pkg/watchers/sui/watcher.go

```diff
@@ -188,17 +205,61 @@ func (e *Watcher) Run(ctx context.Context) error {
 
 			logger.Info("Received obsv request")
 
-		case <-timer.C:
-			if e.unsafeDevMode {
-				logger.Info("tick unsafe")
-			} else {
-				logger.Info("tick safe")
+			id := base64.StdEncoding.EncodeToString(r.TxHash)
+
+			buf := fmt.Sprintf(`{"jsonrpc":"2.0", "id": 1, "method": "sui_getEvents", "params": [{"Transaction": "%s"}, null, 10, true]}`, id)
+
+			logger.Info(buf)
+
+			resp, err := http.Post(e.suiRPC, "application/json", strings.NewReader(buf))
+			if err != nil {
+				logger.Error(e.suiRPC, zap.Error(err))
+				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
+				continue
+			}
+
+			body, err := io.ReadAll(resp.Body)
+			if err != nil {
+				logger.Error(e.suiRPC, zap.Error(err))
+				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDSui, 1)
+				continue
+
+			}
+			logger.Info(string(body))
+
+			if !gjson.Valid(string(body)) {
+				logger.Error("InvalidJson: " + string(body))
+				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDAptos, 1)
+				break
+			}
+
+			outcomes := gjson.ParseBytes(body).Get("result.data")
+			if !outcomes.Exists() {
+				logger.Error("InvalidJson: " + string(body))
+				p2p.DefaultRegistry.AddErrorCount(vaa.ChainIDAptos, 1)
+				break
 			}
+
+			for _, chunk := range outcomes.Array() {
+				err := e.inspectBody(logger, chunk)
+				if err != nil {
+					logger.Error(e.suiRPC, zap.Error(err))
+				}
+			}
+
+		case <-timer.C:
 			for {
 				var msg = make([]byte, 10000)
 				var n int
-				ws.SetReadDeadline(time.Now().Local().Add(100_000_000))
+				err := ws.SetReadDeadline(time.Now().Local().Add(100_000_000))
```

Comment:
not sure we need the local here

Reply:
@jynnantonix you may know better how deadlines in the ws library are handled and whether this is correct handling

Reply:
I agree the `Local()` is not necessary here.  Otherwise as far as I can tell, this call eventually gets forwarded to `setsockopt` on the underlying socket so this should be fine.